src/fenicsxconcrete/finite_element_problem/base_material.py

```diff
@@ -55,34 +56,7 @@ def __init__(
         self.experiment.p = self.p  # update experimental parameter list for use in e.g. boundary definition
 
         # set log level...
-        if self.p["log_level"] == "DEBUG":
-            df.log.LogLevel(10)
-            logging.getLogger("FFC").setLevel(logging.DEBUG)
-            logging.getLogger("UFL").setLevel(logging.DEBUG)
-            logger.add(sys.stderr, level="DEBUG")
-        elif self.p["log_level"] == "INFO":
-            df.log.LogLevel(20)
-            logging.getLogger("FFC").setLevel(logging.INFO)
-            logging.getLogger("UFL").setLevel(logging.INFO)
-            logger.add(sys.stderr, level="INFO")
-        elif self.p["log_level"] == "WARNING":
-            df.log.LogLevel(30)
-            logging.getLogger("FFC").setLevel(logging.WARNING)
-            logging.getLogger("UFL").setLevel(logging.WARNING)
-            logger.add(sys.stderr, level="WARNING")
-        elif self.p["log_level"] == "ERROR":
-            df.log.LogLevel(40)
-            logging.getLogger("FFC").setLevel(logging.ERROR)
-            logging.getLogger("UFL").setLevel(logging.ERROR)
-            logger.add(sys.stderr, level="ERROR")
-        elif self.p["log_level"] == "CRITICAL":
-            df.log.LogLevel(50)
-            logging.getLogger("FFC").setLevel(logging.CRITICAL)
-            logging.getLogger("UFL").setLevel(logging.CRITICAL)
-            logger.add(sys.stderr, level="CRITICAL")
-        else:
-            level = self.p["log_level"]
-            raise ValueError(f"unknown log level {level}")
+        self.logger.setLevel(self.p["log_level"])
```

Comment:
Is there the same in the base experiment?
Then we would have a simple and in my opinion user friendly way of setting the log level for the whole problem.
Testing this would have been what I had in mind for the last test with the different levels. To give a simple example of how to set the different levels.

Reply:
> Is there the same in the base experiment?

I am not really satisfied with what I did here. I am thinking about a solution where the logging can be done on module-level, but also that each class can have a logger attribute.

>  Testing this would have been what I had in mind for the last test with the different levels. To give a simple example of how to set the different levels.

I think I had something like this in the first version .... :laughing:

Reply:
I don't want to make this more complex and a bigger problem than it is. Feel free do decide that you know better and we will just see in the long run if this work for us. 
But while we are still discussing: What do you mean on a module level?
In practice, currently we have the setup and the material. although I don't see why we need to worry about setting different loggers for the different parts, but I am also not familiar with good practice in logging.
On that note, I have noticed, that the cylinder example uses some external module that has the default logging set to "info" which is a bit annoying. When running [this example](https://github.com/BAMresearch/FenicsXConcrete/blob/58-examples-in-documentation/docs/examples/cylinder.py) you will see what I mean. I would expect/suggest to make sure that external packages are set to the same logging level as module that calls them.

Reply:
> But while we are still discussing: What do you mean on a module level?

in the [tutorial](https://docs.python.org/3/howto/logging.html#advanced-logging-tutorial) they state module-level logging is a good convention. With `fenicsxconcrete` you might want to know where the log message came from.
If the loggers follow a naming hierarchy such as `name = self.__class__.__module__` and you do not explicitly set a level for this specific logger of a class, then the level by the `logger = logging.getLogger("fenicsxconcrete")` will be used.
In short, you can do
```python
from fenicsxconcrete import set_log_levels
set_log_levels({"fenicsxconcrete": logging.DEBUG})
```
to set the level for each logger within the package.

I think it would be nice if certain classes have a logger attribute. In the application you write you could use that to log important information.

> On that note, I have noticed, that the cylinder example uses some external module that has the default logging set to "info" which is a bit annoying. When running [this example](https://github.com/BAMresearch/FenicsXConcrete/blob/58-examples-in-documentation/docs/examples/cylinder.py) you will see what I mean. I would expect/suggest to make sure that external packages are set to the same logging level as module that calls them.

I guess this is the output from `dolfinx.io.gmshio.read_from_msh`. 
In general I would rather let the user set the log level for `dolfinx`, but the mesh info can be suppressed.

Reply:
Maybe we need to move this discussion to the in person meeting. I still feel we are talking about different use cases.
And I think there is a way to allow for all the flexibility that you are trying to save while still making it easier for unexperienced people.
The way I want to design the interface for this package, is for people without having any idea what dolphinx is.
We are providing a package where we can develop features that are required within our group, but which need to be shared with other people who are not experts on FEM and are more or less able to use different python packages.
I agree that is is neccessary to have the possibility to define all the things you want but I dont want this to be the only and the default way.
Even for me, I would like to have preferably 3 settings: 1: no output unless something fails, 2: minimum output so that I know something is still going on internally, 3: maximum output when I need to understand whats happening
I dont want to then need to set different outputs or search how the definition of the logger is.

Reply:
> So if I understand you correctly, the only thing is missing, is that the log level for fenicsxconcrete should also be set for dolfinx etc.?

Yes, that would be amazing, maybe likewise with the meshing package (or others that we use and that might have relevant output)